static/js/disaster_dashboard/map_section.tsx

```diff
@@ -0,0 +1,252 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Component for the map section of the disaster dashboard
+ */
+
+import * as d3 from "d3";
+import _ from "lodash";
+import React, { useEffect, useRef } from "react";
+import ReactDOM from "react-dom";
+
+import { addMapPoints, drawD3Map, getProjection } from "../chart/draw_d3_map";
+import { GeoJsonData, GeoJsonFeatureProperties } from "../chart/types";
+import {
+  EUROPE_NAMED_TYPED_PLACE,
+  IPCC_PLACE_50_TYPE_DCID,
+  USA_PLACE_DCID,
+} from "../shared/constants";
+import { NamedPlace } from "../shared/types";
+import { removeSpinner } from "../shared/util";
+import { getAllChildPlaceTypes, getParentPlaces } from "../tools/map/util";
+import { isChildPlaceOf } from "../tools/shared_util";
+import { CONTENT_SPINNER_ID } from "./constants";
+import { InfoCard } from "./info_card";
+import { DisasterEventPoint, DisasterType, PlaceInfo } from "./types";
+
+const MAP_ID = "disaster-dashboard-map";
+const LEGEND_CONTAINER_ID = "disaster-dashboard-legend";
+const ZOOM_IN_BUTTON_ID = "zoom-in-button";
+const ZOOM_OUT_BUTTON_ID = "zoom-out-button";
+const iNFO_CARD_OFFSET = 5;
+
+const DISASTER_COLORS = {
+  [DisasterType.EARTHQUAKE]: "red",
+  [DisasterType.FIRE]: "orange",
+  [DisasterType.STORM]: "yellow",
+  [DisasterType.FLOOD]: "green",
+  [DisasterType.DROUGHT]: "blue",
+};
+
+interface MapSectionPropType {
+  disasterEventPoints: DisasterEventPoint[];
+  geoJson: GeoJsonData;
+  selectedDisaster: DisasterType;
+  selectedDate: string;
+  selectedPlaceInfo: PlaceInfo;
+  selectedIntensityProp: string;
+  onPlaceUpdated: (place: NamedPlace) => void;
+  europeanPlaces: NamedPlace[];
```

Comment:
specifically, it seems these are europeanCountries? how about `fetchedEuropeanPlaces` to distinguish it from constants (which is what one might assume on first read).

please also add comments here, especially for this which is rather unusual

Reply:
done